src/server/GameController.cpp

```diff
@@ -49,9 +61,19 @@ void GameController::computeGameState(
     const std::vector<int>& client_ids,
     const std::vector<client_direction_update>& client_direction_updates
 ) {
+    // Update game objects
     refreshPlayers(client_ids);
     updatePlayers(client_direction_updates);
-    refreshAndUpdateGroups();
+    updateGroups();
```

Comment:
It would be nice if we iterated over all GameObjects and called update rather than doing it individually for Groups, Mines, etc

Reply:
Not sure how to do this yet... we would have to store a vector of GameObjects instead of Groups and Mines.
Will leave for another PR.